src/test/run-make-fulldeps/coverage-reports/expected_show_coverage.conditions.txt

```diff
@@ -38,69 +38,53 @@
    37|       |    }
    38|       |
    39|      1|    if true {
-   40|       |        // Demonstrate the difference with `TerminatorKind::Assert` as of 2020-11-15. Assert is no
-   41|       |        // longer treated as a `BasicCoverageBlock` terminator, which changed the coverage region,
-   42|       |        // for the executed `then` block above, to include the closing brace on line 30. That
-   43|       |        // changed the line count, but the coverage code region (for the `else if` condition) is
-   44|       |        // still valid.
-   45|       |        //
-   46|       |        // Note that `if` (then) and `else` blocks include the closing brace in their coverage
-   47|       |        // code regions when the last line in the block ends in a semicolon, because the Rust
-   48|       |        // compiler inserts a `StatementKind::Assign` to assign `const ()` to a `Place`, for the
-   49|       |        // empty value for the executed block. When the last line does not end in a semicolon
-   50|       |        // (that is, when the block actually results in a value), the additional `Assign` is not
-   51|       |        // generated, and the brace is not included.
-   52|      1|        let mut countdown = 0;
-   53|      1|        if true {
-   54|      1|            countdown = 10;
-   55|      1|        }
-   56|       |
-   57|      1|        if countdown > 7 {
-   58|      1|            countdown -= 4;
-   59|      1|        }
-   60|       |        // The closing brace of the `then` branch is now included in the coverage region, and shown
-   61|       |        // as "executed" (giving its line a count of 1 here). Since, in the original version above,
-   62|       |        // the closing brace shares the same line as the `else if` conditional expression (which is
-   63|       |        // not executed if the first `then` condition is true), only the condition's code region is
-   64|       |        // marked with a count of 0 now.
-   65|      0|        else if countdown > 2 {
-   66|      0|            if countdown < 1 || countdown > 5 || countdown != 9 {
-   67|      0|                countdown = 0;
-   68|      0|            }
-   69|      0|            countdown -= 5;
-   70|       |        } else {
-   71|      0|            return;
-   72|       |        }
-   73|      1|    }
-   74|       |
-   75|      1|    let mut countdown = 0;
-   76|      1|    if true {
-   77|      1|        countdown = 1;
-   78|      1|    }
-   79|       |
-   80|      1|    let z = if countdown > 7 {
+   40|      1|        let mut countdown = 0;
+   41|      1|        if true {
+   42|      1|            countdown = 10;
+   43|      1|        }
+   44|       |
+   45|      1|        if countdown > 7 {
+   46|      1|            countdown -= 4;
+   47|      1|        }
+   48|      0|        else if countdown > 2 {
```

Comment:
With the count on the closing brace above being 1, I'd expect to see 1 here too since the condition must be evaluated.

Reply:
I believe this is correct, but it may be easier to see in `--color`:

<img width="471" alt="Screen Shot 2020-12-01 at 9 59 04 PM" src="https://user-images.githubusercontent.com/3827298/100834357-9ba7df00-3420-11eb-9042-1535abdf293f.png">

The condition on line 45 is `true`, and the block surrounding line 46 is executed; so there's no need to evaluate the condition on line 48.

Reply:
I see, so do we always count the closing brace itself as part of the if body (whether or not it has an `else`?) I was thinking we didn't include the brace, and that the count on that line was for the "condition was false" branch.

Reply:
Yes... and the "condition was false" is indicated by the coverage of the block and statements _after_ the `else` (if any). I **think** this is the most accurate it can be.

This was part of the improvement to get rid of `Goto` special case handling, and as discussed above, the counter carat `^0` for an implicit else appears _after_ the closing brace now _because_ the closing brace is part of the `Block` of the previous `if`.

I also tried to show this by the unorthodox formatting, showing various pieces of the syntax on different lines, occasionally, to document through testing how the coverage covers each part (since the test result don't have the benefit of colors).

Reply:
Got it, thanks.